rust/src/dataset.rs

```diff
@@ -720,17 +721,18 @@ mod tests {
         );
         let expected_batches = batches.batches.clone();
 
-        let test_uri = test_dir.path().to_str().unwrap();
+        let url = Url::from_file_path(test_dir.path()).unwrap();
```

Comment:
is `url_ref` to like `file://`?

does it mean that it will not cover the case with starting `/`?